full-stack-tests/presentation/src/learning-snippets/Hierarchies.test.ts

```diff
@@ -24,305 +28,1671 @@ describe("Learning Snippets", () => {
+          expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+            label: { displayValue: "My Node" },
+          }]);
+        });
+
+        it("uses `description` attribute", async () => {
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.Description.Ruleset
+          // The ruleset has a root node specification that returns a single custom node and assigns it a description.
+          const ruleset: Ruleset = {
+            id: "example",
+            rules: [{
+              ruleType: RuleTypes.RootNodes,
+              specifications: [{
+                specType: ChildNodeSpecificationTypes.CustomNode,
+                type: "T_MY_NODE",
+                label: "My Node",
+                description: "My node's description",
+              }],
+            }],
+          };
+          // __PUBLISH_EXTRACT_END__
+          printRuleset(ruleset);
+
+          // Verify that node with correct description is returned
+          const nodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+          expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+            description: "My node's description",
+          }]);
+        });
+
+        it("uses `imageId` attribute", async () => {
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.ImageId.Ruleset
+          // The ruleset has a root node specification that returns a single custom node and assigns it an image identifier.
+          const ruleset: Ruleset = {
+            id: "example",
+            rules: [{
+              ruleType: RuleTypes.RootNodes,
+              specifications: [{
+                specType: ChildNodeSpecificationTypes.CustomNode,
+                type: "T_MY_NODE",
+                label: "My Node",
+                imageId: "my-icon-identifier",
+              }],
+            }],
+          };
+          // __PUBLISH_EXTRACT_END__
+          printRuleset(ruleset);
+
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.ImageId.Result
+          // Verify that node with correct image identifier is returned
+          const nodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+          expect(nodes).to.have.lengthOf(1).and.to.containSubset([{
+            imageId: "my-icon-identifier",
+          }]);
+          // __PUBLISH_EXTRACT_END__
+        });
+
+        it("uses `hideNodesInHierarchy` attribute", async () => {
+          // __PUBLISH_EXTRACT_START__ Hierarchies.CustomNodeSpecification.HideNodesInHierarchy.Ruleset
+          // The ruleset contains a root node specification for `bis.PhysicalModel` nodes which are grouped by class and hidden. This
+          // means class grouping nodes are displayed, but instance nodes are hidden and instead their children are displayed. The
+          // children are determined by another rule.
```

Comment:
This comment seems to have come from some other ruleset.

```suggestion
          // This ruleset produces a hierarchy that consists of two custom nodes. The parent node is hidden by
          // `hideNodesInHierarchy` attribute, thus its child appears one hierarchy level higher.
```